Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -88,22 +88,22 @@ default SeekableByteChannel getWriteChannel(@NotNull final String path, final bo
     SeekableByteChannel getWriteChannel(@NotNull Path path, boolean append) throws IOException;
 
     /**
-     * Applies the supplied action on the list of file URIs contained inside the given directory. The logic is
-     * non-recursive.
+     * Returns a stream of URIs, the elements of which are the entries in the directory. The listing is non-recursive.
+     * Note that the URIs supplied by the stream will be file URIs (not ending with "/") irrespective of whether the URI
+     * corresponds to a file or a directory. The caller should manage file vs. directory handling in the processor.
      *
-     * @param directoryURI The URI of the directory to list
-     * @param processor A method to apply to the URIs in the directory
+     * @param directory the URI of the directory to list
+     * @return The {@link Stream} of {@link URI}s
      */
-    void applyToChildURIs(@NotNull URI directoryURI, @NotNull Consumer<URI> processor) throws IOException;
-    // TODO where should this code be kept, because it doesn't have anything to do with the channel provider
-    // but it does need the s3 stuff, like how to build a client, etc. that is present here. Also, what should I name
-    // it?
+    Stream<URI> list(@NotNull URI directory) throws IOException;
 
     /**
-     * Applies the supplied action on the list of file URIs contained inside the given directory recursively.
+     * Returns a stream of URIs, the elements of which are all the entries in the directory. The listing is recursive.
```

Comment:
"the elements of which are all the entries in the directory" - this doesn't seem correct for a "walk".